mkosi/__init__.py

```diff
@@ -3607,13 +3634,116 @@ def insert_verity(
             loopdev,
             args.verity_partno,
             verity,
-            root_partition_name(args, True),
+            root_partition_name(args, "Verity"),
             gpt_root_native(args.architecture, args.usr_only).verity,
             True,
             u,
         )
 
 
+PKCS7_NOCERTS  = 0x2    # Don't include signature certificates in result
+PKCS7_DETACHED = 0x40   # Don't include data to sign in result
+PKCS7_BINARY   = 0x80   # Don't mangle newlines for MIME canonical format
+PKCS7_NOATTR   = 0x100  # Don't include signature time, … in result
+
+
+def make_verity_sig(
+    args: CommandLineArguments, root_hash: Optional[str], do_run_build_script: bool, for_cache: bool
+) -> Tuple[Optional[BinaryIO], Optional[bytes], Optional[str]]:
+
+    if do_run_build_script or args.verity != "signed":
+        return None, None, None
+    if for_cache:
+        return None, None, None
+
+    assert root_hash is not None
+    assert crypto is not None
```

Comment:
shouldn't this return and print an error, instead of asserting?

Reply:
well, so far if any of our deps are missing (as in: binaries we want to call) we just fail with an uncaught exception, so my idea was here to expose the same behaviour for the openssl lib missing, after all this will trigger an uncaught exception here too.

Reply:
Please just return an error. That's much more user-friendly.

Reply:
I think we might have this in a few other places as well and it is a potential footgun. python as a `-O` option (and `PYTHONOPTIMIZE` environment variable) that throws away asserts. I know that at least Arch builds its Python packages with this by default.

Reply:
On a related note, I think it's still better to move the import into this function and die with a nice error message on `ImportError`, since mypy doesn't like a potential module object being `None`. 

There's probably a nicer solution for this, but since this function is the only user of this module - at least for now -  I think this is a clean solution.

Reply:
@keszybz What do you mean "return an error"? you mean call `die()` with an explanatory message?